clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -928,39 +927,94 @@ public String toRst() {
+
+            if (def.group != null) {
+                if (!lastKeyGroupName.equalsIgnoreCase(def.group)) {
+                    b.append(def.group);
+                    b.append("\n");
+                    char[] underLine = new char[def.group.length()];
+                    Arrays.fill(underLine, '^');
+                    b.append(new String(underLine));
+                    b.append("\n\n");
                 }
-                b.append("  ");
-                b.append(docLine);
-                b.append("\n\n");
+                lastKeyGroupName = def.group;
             }
-            b.append("  * Type: ");
-            b.append(def.type.toString().toLowerCase(Locale.ROOT));
-            b.append("\n");
-            if (def.defaultValue != null) {
-                b.append("  * Default: ");
-                if (def.type == Type.STRING) {
-                    b.append("\"");
-                    b.append(def.defaultValue);
-                    b.append("\"");
-                } else {
-                    b.append(def.defaultValue);
+
+            b.append(getConfigKeyDetails(def));
+
+            if (def.dependents != null && def.dependents.size() > 0) {
+                int j = 0;
+                b.append("  * Dependents: ");
+                for (String dependent : def.dependents) {
+                    b.append("``");
+                    b.append(dependent);
+                    if (++j == def.dependents.size())
+                        b.append("``");
+                    else
+                        b.append("``,");
                 }
                 b.append("\n");
             }
-            b.append("  * Importance: ");
-            b.append(def.importance.toString().toLowerCase(Locale.ROOT));
-            b.append("\n\n");
+            b.append("\n");
         }
         return b.toString();
     }
 
     /**
+     * Shared content on Rst and Enriched Rst.
+     */
+    private StringBuilder getConfigKeyDetails(ConfigKey def) {
+        StringBuilder b = new StringBuilder();
+
+        b.append("``");
+        b.append(def.name);
+        b.append("``\n");
+        for (String docLine : def.documentation.split("\n")) {
+            if (docLine.length() == 0) {
+                continue;
+            }
+            b.append("  ");
+            b.append(docLine);
+            b.append("\n\n");
+        }
+        b.append("  * Type: ");
+        b.append(def.type.toString().toLowerCase(Locale.ROOT));
+        b.append("\n");
+        if (def.defaultValue != null) {
+            b.append("  * Default: ");
+            if (def.type == Type.STRING) {
+                b.append("\"");
+                b.append(def.defaultValue);
+                b.append("\"");
+            } else {
+                b.append(def.defaultValue);
+            }
+            b.append("\n");
+        }
+        b.append("  * Importance: ");
+        b.append(def.importance.toString().toLowerCase(Locale.ROOT));
+        b.append("\n\n");
```

Comment:
The second newline should be left for the caller, as it otherwise causes an extra line before 'Dependents' in the enriched RST